### PR TITLE
Add default values to external APIs

### LINKED ISSUE #336: allow basic defaulting of some external api items
At the moment, validation requires a routerprofile to be provided as input on cluster creation:
```
"routerProfiles": [
  {
    "name": "default"
  }
]
```

If the user is not providing a publicSubdomain it is a bit pointless to have to provide such a routerProfile - we could default that value appropriately.

### PR DESCRIPTION
**Pull request by ehashman**
```release-note
Set some default values (e.g. RouterProfile) when not specified upon creating a new cluster config
```
Addresses #336.

- [x] Default RouterProfiles.
- [ ] Default masterPoolProfile/count=3
- [ ] Default agentPoolProfiles[name=infra]/count=3
- [ ] Default agentPoolProfiles[*]/osType=Linux
- [x] Unit test defaults
- [ ] Simplify test/manifests/fakerp/create.yaml

### CODE REVIEW

**Comment by ehashman** [comment]
/retest

flake #1549

File: pkg/api/2018-09-30-preview/default.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package v20180930preview
+
+func setDefaults(oc *OpenShiftManagedCluster) {
+	defaultRouterName := "default"
```
Comment (kwoodson) [inline pkg/api/2018-09-30-preview/default.go]: default router name is `router`

File: pkg/api/2019-04-30/default.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package v20190430
+
+func setDefaults(oc *OpenShiftManagedCluster) {
+	defaultRouterName := "default"
```
Comment (kwoodson) [inline pkg/api/2019-04-30/default.go]: same as above

File: pkg/api/2018-09-30-preview/default.go (hunk 1 above)
Comment (ehashman) [reply pkg/api/2018-09-30-preview/default.go]: Hm, are you sure? I got this from https://github.com/openshift/openshift-azure/blob/a4338bc5aeaff48ba2a37bec28ae249231f47dd8/pkg/fakerp/fakerp.go#L275-L281

I'd be concerned if I'm setting the wrong name here while all the tests are passing...

Comment (jim-minter) [reply pkg/api/2018-09-30-preview/default.go]: @ehashman you're right

File: pkg/api/2018-09-30-preview/default.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+	if len(oc.Properties.RouterProfiles) == 0 {
+		oc.Properties.RouterProfiles = []RouterProfile{
+			{
+				Name: &defaultRouterName,
```
Comment (jim-minter) [inline pkg/api/2018-09-30-preview/default.go]: in our codebase prefer using to.StringPtr() to skip the extra variable definition

**Comment by jim-minter** [comment]
good start, cross-check against https://github.com/openshift/openshift-azure/issues/336 and https://github.com/openshift/openshift-azure/blob/master/test/manifests/fakerp/create.yaml for other obvious defaultables.  I think the following are certainly plausible:

masterPoolProfile/count=3
agentPoolProfiles[name=infra]/count=3
agentPoolProfiles[*]/osType=Linux

+ unit test
+ simplify /test/manifests/fakerp/create.yaml

File: pkg/api/2018-09-30-preview/default.go (hunk 1 above)
Comment (kwoodson) [reply pkg/api/2018-09-30-preview/default.go]: I thought this was referring to the actual router name instead of the routerprofile name.

**Comment by ehashman** [comment]
@jim-minter: added your suggestions to a TODO list in the first comment.

In 5908786f2a20cf3195035efed508e22210c0601d I added unit tests and that's taken the PR size from medium to large; perhaps we should split the remaining items into follow-up PRs? Namely, one for more defaults for the master/agentpool profiles, and another for simplifying the test manifest. (If you agree, feel free to remove the WIP label.)

**Comment by ehashman** [comment]
Will split the other TODOs into two follow-up PRs.

/retest

**Pull request merged**